src/twisted/test/proto_helpers.py

```diff
@@ -74,10 +81,20 @@ def connectionLost(self, reason):
             d.callback(None)
 
 
+
+deprecatedModuleAttribute(
+    Version("Twisted", 19, 3, 0),
+    "Please use twisted.internet.testing.AccumulatingProtocol instead",
```

Comment:
I think it would make more sense to deprecate `proto_helpers` in its entirety, rather than every individual name?  You can put the `deprecatedModuleAttribute` at the package level (in `__init__.py`) instead, so you have fewer of these to spell out.

In either case, you should remove all the duplicated code from this module, and re-export the names via `__all__`, so that we don't have them all copied twice until their removal, and we don't get divergent behavior.

Reply:
Alright, moved the deprecation warning code to twisted.test.__init__.